src/axom/quest/detail/MarchingCubesImpl.hpp

```diff
@@ -0,0 +1,998 @@
+  #else
+  using ScanPolicy = typename axom::execution_space<ExecSpace>::loop_policy;
+  #endif
+#endif
+  using SequentialLoopPolicy =
+    typename execution_space<SequentialExecSpace>::loop_policy;
+  static constexpr auto MemorySpace = execution_space<ExecSpace>::memory_space;
+
+  AXOM_HOST MarchingCubesImpl(int allocatorID,
+                              axom::Array<std::uint16_t>& caseIdsFlat,
+                              axom::Array<std::uint16_t>& crossingFlags,
+                              axom::Array<axom::IndexType>& scannedFlags,
+                              axom::Array<std::uint16_t>& facetIncrs)
+    : m_allocatorID(allocatorID)
+    , m_caseIds()
+    , m_caseIdsIndexer()
+    , m_caseIdsFlat(caseIdsFlat)
+    , m_crossingCases(0, 0, m_allocatorID)
+    , m_crossingFlags(crossingFlags)
+    , m_scannedFlags(scannedFlags)
+    , m_crossingParentIds(0, 0, m_allocatorID)
+    , m_facetIncrs(facetIncrs)  // (0, 0, m_allocatorID)
+    , m_firstFacetIds(0, 0, m_allocatorID)
+  {
+    SLIC_ASSERT(caseIdsFlat.getAllocatorID() == allocatorID);
+    SLIC_ASSERT(crossingFlags.getAllocatorID() == allocatorID);
+    SLIC_ASSERT(scannedFlags.getAllocatorID() == allocatorID);
+  }
+
+  /*!
+    @brief Initialize data to a blueprint domain.
+    @param dom Blueprint structured mesh domain
+    @param topologyName Name of mesh topology (see blueprint
+           mesh documentation)
+    @param maskFieldName Name of integer cell mask function is in dom
+
+    Set up views to domain data and allocate other data to work on the
+    given domain.
+
+    The above data from the domain MUST be in a memory space
+    compatible with ExecSpace.
+  */
+  AXOM_HOST void setDomain(const conduit::Node& dom,
+                           const std::string& topologyName,
+                           const std::string& maskFieldName) override
+  {
+    // Time this due to potentially slow memory allocation
+    AXOM_PERF_MARK_FUNCTION("MarchingCubesImpl::initialize");
+    clearDomain();
+
+    SLIC_ASSERT(conduit::blueprint::mesh::topology::dims(dom.fetch_existing(
+                  axom::fmt::format("topologies/{}", topologyName))) == DIM);
+
+    m_mvu = axom::quest::MeshViewUtil<DIM, MemorySpace>(dom, topologyName);
+
+    m_bShape = m_mvu.getCellShape();
+    m_coordsViews = m_mvu.getConstCoordsViews(false);
+    if(!maskFieldName.empty())
+    {
+      m_maskView = m_mvu.template getConstFieldView<int>(maskFieldName, false);
+    }
+  }
+
+  AXOM_HOST void setDataParallelism(MarchingCubesDataParallelism dataPar) override
+  {
+    constexpr MarchingCubesDataParallelism autoPolicy =
+      std::is_same<ExecSpace, axom::SEQ_EXEC>::value
+      ? MarchingCubesDataParallelism::hybridParallel
+      :
+#if defined(AXOM_USE_OPENMP) && defined(AXOM_USE_RAJA)
+      std::is_same<ExecSpace, axom::OMP_EXEC>::value
+        ? MarchingCubesDataParallelism::hybridParallel
+        :
+#endif
+        MarchingCubesDataParallelism::fullParallel;
```

Comment:
Minor (readability): 

Should the "else" case of the ternary operators (`:`) move into the `#ifdef` e.g.
```suggestion
    constexpr MarchingCubesDataParallelism autoPolicy =
      std::is_same<ExecSpace, axom::SEQ_EXEC>::value
      ? MarchingCubesDataParallelism::hybridParallel
#if defined(AXOM_USE_OPENMP) && defined(AXOM_USE_RAJA)
      : std::is_same<ExecSpace, axom::OMP_EXEC>::value
        ? MarchingCubesDataParallelism::hybridParallel
#endif
        : MarchingCubesDataParallelism::fullParallel;
```

Reply:
Ok.  Done.